core-common/src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -173,6 +173,8 @@ public final void registerSession(final Session session) {
     public final void unregisterSession(final Session session) {
         try {
             getCurator().delete().guaranteed().forPath(getSubscriptionPath("/sessions/" + session.getId()));
```

Comment:
@a1exsh What does it mean - guaranteed() ?

Reply:
Well, from the docs: `Solves edge cases where an operation may succeed on the server but connection failure occurs before a response can be successfully returned to the client.`
https://curator.apache.org/apidocs/org/apache/curator/framework/api/Guaranteeable.html

Are you hinting that it should already include ignoring `NoNodeException`?

Reply:
it looks like that doc is more precise https://curator.apache.org/apidocs/org/apache/curator/framework/api/GuaranteeableDeletable.html

Reply:
I only see "wreak havoc" :D